src/metrics-logger.ts

```diff
@@ -0,0 +1,50 @@
+'use strict';
+import { Request } from './custom-typings';
+import { Route, Response, Server } from 'restify';
+import { StatsD, StatsDOptions } from './statsd';
+import * as _ from 'lodash';
+
+type HighResolutionTime = [number, number]; // [seconds, nanoseconds]
+
+export const DEFAULT_KEY_NAME: string = 'handler-0';
+
+export function register(server: Server, statsDOptions: StatsDOptions) {
+    let statsD: StatsD = new StatsD(statsDOptions);
+    let logger = new LogFactory(statsD).createLogger();
+
+    server.on('after', (request: Request, response: Response, route: Route, error) => {
+        logger(request, route);
+    });
+}
+
+export class LogFactory {
+    constructor(private statsD: StatsD) {
+    }
+
+    createLogger() {
+        return (request: Request, route: Route) => {
+            let key: string;
+
+            let timer = _.find(request.timers, (item) => {
```

Comment:
find is available from typescript. we dont really need to bring in lodash for this.